samples/debugmalloc/README.md

```diff
@@ -0,0 +1,328 @@
+# The debug malloc sample
+
+- Introduction about debug malloc.
+- Demonstrate how to enable debug malloc through compiling options.
+- explanation and sample of debug malloc public apis.
+
+## Introduction
+
+Debug malloc is a tool that helps identifying memory leaks in enclaves. The
+main logic of debug malloc is as follows.
+
+- Whenever memory is allocated, debug malloc captures the callstack and
+associates it with the allocated memory. Additionally, debug malloc sets the
+contents of the allocated memory to zero.
+
+- Whenever memory is freed, it clears the contents to a fixed pattern to help
```

Comment:
```suggestion
- Whenever memory is freed, all bytes are set to 0x0D to help
```
Explain to the developer what the fixed pattern is, since they will need to be able to recognize it in debugging